<launch>
  <arg name="nodelet" default="false"/>

  <!-- Launch nodelet -->
  <node if="$(arg nodelet)" pkg="nodelet" type="nodelet" name="realtime_urtf_filter_nodelet" args="load realtime_urdf_filter/RealtimeURDFFilterNodelet camera_nodelet_manager" output="screen">
    <remap from="~input_depth" to="/panda/depth_camera/depth_image"/>
    <remap from="~output_depth" to="/panda/depth_camera/depth_image/filtered" />
    <remap from="~output_mask" to="/urdf_filtered_mask" />

    <rosparam command="load" file="$(find realtime_urdf_filter)/launch/filter_parameters.yaml"/>
  </node>                 

  <node unless="$(arg nodelet)" 
    pkg="realtime_urdf_filter" type="realtime_urdf_filter" name="$(anon realtime_urdf_filter)" output="screen">
    <remap from="~input_depth" to="/panda/depth_camera/depth_image"/>
    <remap from="~output_depth" to="/panda/depth_camera/depth_image/filtered" />
    <remap from="~output_mask" to="/urdf_filtered_mask" />

    <rosparam command="load" file="$(find realtime_urdf_filter)/launch/filter_parameters.yaml"/>
  </node>                 

  <!-- Load an example URDF -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find franka_description)/robots/panda_arm_hand.urdf.xacro'"/>
  
  <!-- Produce Point Cloud -->
  
  <node pkg="nodelet" type="nodelet" args="manager" name="nodelet_manager" output="screen"/>
  
  <node pkg="realtime_urdf_filter" name="nanifier" type="nanify.py" output="screen"> 
    <rosparam command="load" file="$(find realtime_urdf_filter)/launch/filter_parameters.yaml"/>
  </node>
  
  <node pkg="nodelet" type="nodelet" name="cloudify" args="load depth_image_proc/point_cloud_xyz nodelet_manager">
    <remap from="image_rect" to="panda/depth_camera/depth_image/filtered/image"/> <!-- image input -->
    <remap from="points" to="panda/depth_camera/depth_image/filtered/points"/> <!-- pointcloud output -->
    <remap from="camera_info" to="panda/depth_camera/depth_image/filtered/camera_info"/> <!-- camera info -->
  </node>

<!--
  <node name="robot_state_publisher_urdf_filter" pkg="robot_state_publisher" type="state_publisher">
    <param name="tf_prefix" value=""/>
  </node>
-->
</launch>

